akka-actor/src/main/scala/akka/pattern/BackoffOnStopSupervisor.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2018-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import akka.actor.SupervisorStrategy.{ Directive, Escalate }
+import akka.actor.{ Actor, ActorLogging, OneForOneStrategy, Props, SupervisorStrategy, Terminated }
+import akka.annotation.InternalApi
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * INTERNAL API
+ *
+ * Back-off supervisor that stops and starts a child actor using a back-off algorithm when the child actor stops.
+ * This back-off supervisor is created by using `akka.pattern.BackoffSupervisor.props`
+ * with `BackoffOpts.onStop`.
+ */
+@InternalApi private[akka] class BackoffOnStopSupervisor(
```

Comment:
BackoffOnStopSupervisor is `private[akka]` for backwards compability, the other `BackoffOnRestartSupervisor` is just `private`. Moving to internal package would result in opening the latter on up to `private[akka]`

Reply:
I don't get how that does anything for backward compatibility, there was no `BackoffOnStopSupervisor` before, and `BackoffOnRestartSupervisor` was private already, so it should be fine to have those in a more clear no-go package.

Reply:
`BackoffOnStopSupervisor` is `private[akka]` and not private because it's extended by the `BackoffSupervisor` which is required for compatibility.
Moving them to an internal package would make one class more obviously internal, while the other is opened up. I can move it but it's kind of win/lose :)